### PR TITLE
Add Tuple[slice] in iLocIndexerFrame. Fix #143

### LINKED ISSUE #143: [BUG] Argument of type "tuple[slice]" cannot be assigned to parameter "idx"
**Describe the bug**
False positive error warning when using `iloc` function.

**To Reproduce**
Using pyright on
```
import pandas as pd

df = pd.DataFrame({'Char': ['A', 'B', 'C'], 'Number': [1, 2, 3]})
df = df.iloc[0:2,]
```
produces
```
Argument of type "tuple[slice]" cannot be assigned to parameter "idx" of type "IndexType | MaskType | Tuple[IndexType | MaskType, IndexType | MaskType]" in function "__getitem__"    Type "tuple[slice]" cannot be assigned to type "IndexType | MaskType | Tuple[IndexType | MaskType, IndexType | MaskType]"      "tuple[slice]" is incompatible
```

**Please complete the following information:**
 - OS: ArchLinux
 - python 3.10.5
 - pyright 1.1.260
 - pandas-stub 1.4.3.220704

### PR DESCRIPTION
**Pull request by AchmadFathoni**
- [x] Closes #143
- [x] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

**Comment by bashtage** [comment]
Is that an actual supported mode for iloc? Or just an accident that works?

https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.iloc.html

**Comment by AchmadFathoni** [comment]
I don't know, I'm out of deep in this topic. I just following @Dr-Irv suggestion.

**Comment by Dr-Irv** [comment]
> Is that an actual supported mode for iloc? Or just an accident that works because tuples are unpacked in `__s/getitem__`?
> 
> https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.iloc.html

The example in the issue is `df.iloc[0:2,]`.  Here the arguments to `__getitem__()` is `tuple(slice,)`.  Agreed it's an odd usage, but valid.

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
Can you also add a test in `test_frame.py` that fails without this change, and works with it?  Copying the code from the original issue will suffice.  Make sure to include a comment that refers to the issue.

**Comment by bashtage** [comment]
> The example in the issue is `df.iloc[0:2,]`. Here the arguments to `__getitem__()` is `tuple(slice,)`. Agreed it's an odd usage, but valid.

I agree it works.  I am just querying if pandas-stubs should be adding types that aren't part of the docstring. Shouldn't there be a change upstream to confirm that this is a supported format?

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
CI failing due to `black`.  Do `poe style` to make sure everything is OK.

File: pandas-stubs/core/frame.pyi [hunk 1]
```diff
@@ -89,7 +89,10 @@ class _iLocIndexerFrame(_iLocIndexer):
     @overload
     def __getitem__(
         self,
-        idx: IndexType | MaskType | tuple[IndexType | MaskType, IndexType | MaskType] | tuple[slice],
+        idx: IndexType
+        | MaskType
+        | tuple[IndexType | MaskType, IndexType | MaskType]
+        | tuple[slice],
```
Comment (Dr-Irv) [inline pandas-stubs/core/frame.pyi:95]: Sorry - I should have picked up on this before.  Use 
```python
tuple[IndexType | MaskType, IndexType | MaskType | slice]
```

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

File: pandas-stubs/core/frame.pyi (hunk 1 above)
Comment (AchmadFathoni) [reply pandas-stubs/core/frame.pyi:95]: That change fails the test I made
```
Type "tuple[slice]" cannot be assigned to type "IndexType | MaskType | tuple[IndexType | MaskType, IndexType | MaskType]"
```

Comment (Dr-Irv) [reply pandas-stubs/core/frame.pyi:95]: My mistake.  I missed the comma

**Review by Dr-Irv — APPROVED** [review]
Thanks @AchmadFathoni

**Pull request merged**